benchmarks/Csharp/Benchmarks/NBody.cs

```diff
@@ -0,0 +1,195 @@
+﻿namespace Harness.Benchmarks;
+
+public sealed class NBody : IBenchmark
+{
+    public bool Benchmark(int innerIterations)
+    {
+        var system = new NBodySystem();
+        for (var i = 0; i < innerIterations; i++)
+        {
+            system.Advance(0.01);
+        }
+
+        return VerifyResult(system.Energy(), innerIterations);
+    }
+    
+    private static bool VerifyResult(double result, int innerIterations)
+    {
+        if (innerIterations == 250000)
+        {
+            return result == -0.1690859889909308;
+        }
+        if (innerIterations == 1)
+        {
+            return result == -0.16907495402506745;
+        }
+        Console.WriteLine("No verification result for " + innerIterations + " found");
+        Console.WriteLine("Result is: " + result);
+        return false;
+    }
+}
+
+public sealed class NBodySystem
+{
+    private readonly Body[] bodies;
+
+    public NBodySystem()
+    {
+        bodies = CreateBodies();
+    }
+
+    public static Body[] CreateBodies()
+    {
+        Body[] bodies = new[] {Body.Sun,
+                            Body.Jupiter,
+                            Body.Saturn,
+                            Body.Uranus,
+                            Body.Neptune};
+
+        var px = 0.0;
+        var py = 0.0;
+        var pz = 0.0;
+
+        foreach (var b in bodies)
+        {
+            px += b.Vx * b.Mass;
+            py += b.Vy * b.Mass;
+            pz += b.Vz * b.Mass;
+        }
+
+        bodies[0].OffsetMomentum(px, py, pz);
+
+        return bodies;
+    }
+
+    public void Advance(double dt)
+    {
+        for (var i = 0; i < bodies.Length; i++)
+        {
+            for (var j = i + 1; j < bodies.Length; j++)
+                CalculateVelocities(bodies[i], bodies[j]);
+        }
+
+        foreach (var body in bodies)
+        {
+            body.X += dt * body.Vx;
+            body.Y += dt * body.Vy;
+            body.Z += dt * body.Vz;
+        }
+
+        void CalculateVelocities(Body a, Body b)
```

Comment:
Is there a reason to not follow the code precisely? Any deviation will invalidate any comparison.

Reply:
Look like my refactoring habit took over ;-)
Will change this